akka-stream/src/main/scala/akka/stream/ActorMaterializer.scala

```diff
@@ -39,7 +39,7 @@ object ActorMaterializer {
     val system = actorSystemOf(context)
 
     val settings = materializerSettings getOrElse ActorMaterializerSettings(system)
-    apply(settings, namePrefix.getOrElse("flow"))(context)
+    apply(settings, namePrefix.filter(_.trim.nonEmpty).getOrElse("flow"))(context)
```

Comment:
I think that should rather be an assert that if it is defined it shouldn't be empty?

Reply:
should live the `trim` here.
prefer an `Assertion`

Reply:
To add some rationale, if the user specified a `Some(string)` they likely intended it to contain something, so if that something is empty it sounds like they'd like to know as early as possible.

Reply:
Agreed. From a user's perspective I'd expect to be notified when a argument is not a valid input rather than see future discrepancies with the value provided.